jgrapht-core/src/main/java/org/jgrapht/alg/cycle/HawickJamesSimpleCycles.java

```diff
@@ -213,36 +188,15 @@ private void unblock(Integer u)
         for (int wPos = 0; wPos < B[u].size(); wPos++) {
             Integer w = B[u].get(wPos);
 
-            wPos -= removeFromList(B[u], w);
+            wPos -= frequency(B[u], w);
```

Comment:
The old removeFromList code does the counting and removal in a single pass.  Your new code requires two passes.  I think you can eliminate the frequency call by subtracting the B[u].size before and after the removeAll.

Reply:
Done, thanks:)